readthedocs/projects/tasks.py

```diff
@@ -670,6 +670,8 @@ def run_setup(self, record=True):
 
         Return True if successful.
         """
+        api_v2.build(self.build['id']).reset.post()
```

Comment:
Seems we're calling this on every build? It would be best to run it only in the exception handler, but I'm not 100% sure that code will run properly, so this probably makes sense 👍

Reply:
Thinking more, it's probably fine to have it here. There are likely other cases where we want to clear out old build commands (eg. retrying a build if we implement that), so this is probably the most explicit way of handling it.

Reply:
Also it would be weird for all commands to disappear till the build is re-triggered. But maybe I can check if the build has any commands `build['commands']` before calling the API, let me know if that's better

Reply:
But I'm not sure if a build with no commands can have other meta-data that could confuse the build